spring-data-relational/src/main/java/org/springframework/data/relational/repository/query/RelationalExampleMapper.java

```diff
@@ -0,0 +1,150 @@
+
+import static org.springframework.data.domain.ExampleMatcher.*;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+import org.springframework.data.domain.Example;
+import org.springframework.data.mapping.PersistentPropertyAccessor;
+import org.springframework.data.mapping.PropertyHandler;
+import org.springframework.data.mapping.context.MappingContext;
+import org.springframework.data.relational.core.mapping.RelationalPersistentEntity;
+import org.springframework.data.relational.core.mapping.RelationalPersistentProperty;
+import org.springframework.data.relational.core.query.Criteria;
+import org.springframework.data.relational.core.query.Query;
+import org.springframework.data.support.ExampleMatcherAccessor;
+import org.springframework.util.Assert;
+
+/**
+ * Transform an {@link Example} into a {@link Query}.
+ *
+ * @author Greg Turnquist
+ */
+public class RelationalExampleMapper {
+
+	private final MappingContext<? extends RelationalPersistentEntity<?>, ? extends RelationalPersistentProperty> mappingContext;
+
+	public RelationalExampleMapper(
+			MappingContext<? extends RelationalPersistentEntity<?>, ? extends RelationalPersistentProperty> mappingContext) {
+		this.mappingContext = mappingContext;
+	}
+
+	/**
+	 * Use the {@link Example} to extract a {@link Query}.
+	 * 
+	 * @param example
+	 * @return query
+	 */
+	public <T> Query getMappedExample(Example<T> example) {
+		return getMappedExample(example, mappingContext.getRequiredPersistentEntity(example.getProbeType()));
+	}
+
+	/**
+	 * Transform each property of the {@link Example}'s probe into a {@link Criteria} and assemble them into a
+	 * {@link Query}.
+	 * 
+	 * @param example
+	 * @param entity
+	 * @return query
+	 */
+	private <T> Query getMappedExample(Example<T> example, RelationalPersistentEntity<?> entity) {
+
+		Assert.notNull(example, "Example must not be null!");
+		Assert.notNull(entity, "RelationalPersistentEntity must not be null!");
+
+		PersistentPropertyAccessor<T> propertyAccessor = entity.getPropertyAccessor(example.getProbe());
+		ExampleMatcherAccessor matcherAccessor = new ExampleMatcherAccessor(example.getMatcher());
+
+		final List<Criteria> criteriaBasedOnProperties = new ArrayList<>();
+
+		entity.doWithProperties((PropertyHandler<RelationalPersistentProperty>) property -> {
+
+			if (matcherAccessor.isIgnoredPath(property.getName())) {
+				return;
+			}
+
+			Optional<?> optionalConvertedPropValue = matcherAccessor //
+					.getValueTransformerForPath(property.getName()) //
+					.apply(Optional.ofNullable(propertyAccessor.getProperty(property)));
+
+			// If the value is empty, don't try to match against it
+			if (!optionalConvertedPropValue.isPresent()) {
+				return;
+			}
+
+			Object convPropValue = optionalConvertedPropValue.get();
+			boolean ignoreCase = matcherAccessor.isIgnoreCaseForPath(property.getName());
+
+			String columnName = property.getColumnName().getReference();
```

Comment:
Nit: We can use property names directly as these get translated anyway by a `QueryMapper`.

Reply:
Fixed.